File: section-7-replica-sets/vid-05-load-balance/read_preference.py
#!/usr/bin/env python
from pymongo import MongoClient, ReadPreference
import ssl

def main():
    # open connection and authenticate as read only user
    client = MongoClient('mongodb://localhost:27017/statsdb',
                         read_preference=ReadPreference.SECONDARY)

    # get handle for statsdb and statistics collection
    db = client.statsdb
    collection = db.statistics
    
    # retrieve and output one document
    print collection.find_one()

if __name__ == '__main__':
    main()

File: section-7-replica-sets/vid-05-load-balance/read_preference.php
#!/usr/bin/php
<?php

// login and authenticate as read only user
$m = new MongoClient("mongodb://localhost:27017/statsdb");

// Prefer the nearest server with no tag preference
$m->setReadPreference(MongoClient::RP_NEAREST, array());

// get handle for statsdb and statistics collection
$db = $m->statsdb;
$statistics = $db->statistics;

// retrieve a record and display it
$statistical_record = $statistics->findOne();
var_dump($statistical_record);

?>
